template/data/src/main/java/co/nimblehq/template/data/repository/RepositoryImpl.kt

```diff
@@ -4,10 +4,19 @@ import co.nimblehq.template.data.response.toModels
 import co.nimblehq.template.data.service.ApiService
 import co.nimblehq.template.domain.model.Model
 import co.nimblehq.template.domain.repository.Repository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
 
 class RepositoryImpl constructor(
     private val apiService: ApiService
 ) : Repository {
 
-    override suspend fun getModels(): List<Model> = apiService.getResponses().toModels()
+    override fun getModels(): Flow<List<Model>> = flow {
+        try {
+            val result = apiService.getResponses().toModels()
+            emit(result)
+        } catch (e: Exception) {
+            throw e
```

Comment:
No biggie, but better to name it more clearly.
```suggestion
        } catch (exception: Exception) {
            throw exception
```

Reply:
@luongvo Quick question, why did we wrap this in a try catch if we immediately throw the caught exception anyway? 🤔

Reply:
@ryan-conway if we throw it, we need to catch it at usecase or VM level.
- Catching at usecase level: it's hard for us to check the exception's body such as `HttpException` which can not access from **domain** module. Please checkout more insights here https://github.com/nimblehq/android-templates/issues/213#issuecomment-1135375842. 👋 


- Catching at VM level: we need to have 1 more try/catch to wrap the Flow collector. It's a bad idea I think

```
try{
              useCase.execute()
                .catch {
                   // catch error
                }
                .collect { 
                    // get value
                }
}catch (e: Exception){
   // catch error
}
```

Reply:
@luongvo Ah, so this is just preparation for the custom error mapping, right? 😄

Reply:
@ryan-conway exactly, we will have something like this

![image](https://user-images.githubusercontent.com/16315358/187591106-73be40d3-bf95-4f65-b0bc-f04f7ced8884.png)

Reply:
better add TODO

Reply:
@Wadeewee Fixed in https://github.com/nimblehq/android-templates/pull/292/commits/0d2563c9390e5a2bdfc1ae122af7d966f7093594